### PR TITLE
Added support for buildingQuerySuggest to CommerceQuery

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/JSUI-3218

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: unitTests/ui/CommerceQueryTest.ts [hunk 1]
```diff
@@ -6,41 +6,53 @@ import { IChangeableAnalyticsDataObject, IChangeAnalyticsCustomDataEventArgs } f
...
 
-      it('should add a context value with the listing option', function() {
+      it('should add a context value with the listing option', () => {
         const simulation = Simulate.query(test.env);
 
         expect(simulation.queryBuilder.build().context['listing']).toEqual(test.cmp.options.listing);
       });
 
-      it('should unset tab after resetting the listing value', function() {
+      it('should update the tab in query suggest to the lisitng option', () => {
```
Comment (ThibodeauJF) [inline unitTests/ui/CommerceQueryTest.ts:30]: listing*

File: unitTests/ui/CommerceQueryTest.ts [hunk 2]
```diff
@@ -6,41 +6,53 @@ import { IChangeableAnalyticsDataObject, IChangeAnalyticsCustomDataEventArgs } f
...
+        const simulation = Simulate.querySuggest(test.env, 'abc');
+
+        expect(simulation.buildingQuerySuggestArgs.payload.tab).toEqual(test.cmp.options.listing);
+      });
+
+      it('should add a context value to query suggest with the lisitng option', () => {
```
Comment (ThibodeauJF) [inline unitTests/ui/CommerceQueryTest.ts:36]: listing

File: src/ui/CommerceQuery/CommerceQuery.ts [hunk 3]
```diff
@@ -54,6 +55,7 @@ export class CommerceQuery extends Component {
     this.options = ComponentOptions.initComponentOptions(element, CommerceQuery, options);
 
     this.bind.onRootElement(QueryEvents.doneBuildingQuery, this.handleDoneBuildingQuery);
+    this.bind.onRootElement(OmniboxEvents.buildingQuerySuggest, this.handleBuildingQuerySuggest);
```
Comment (samisayegh) [inline src/ui/CommerceQuery/CommerceQuery.ts:58]: Could there be a situation the standalone search box is configured without query suggest?
If so, would the `buildQuerySuggest` event still be triggered in this case?

If not, is there another event we could use? Or should we add a new event for stand-alone search box interfaces that is guaranteed to be triggered?

**Review by samisayegh — APPROVED** [review]

File: src/ui/CommerceQuery/CommerceQuery.ts (hunk 3 above)
Comment (btaillon) [reply src/ui/CommerceQuery/CommerceQuery.ts:58]: I just tested and the listing is added to the `plan` request sent by the standalone searchbox.

**Pull request merged**